.github/workflows/ci.yml

```
@@ -353,6 +357,96 @@ jobs:
           set -x
           [ "$OLD_RUSTDOC_VERSION" != "$NEW_RUSTDOC_VERSION" ] || exit 1
 
+  cross-feature-caching:
+    # Ensure that cached rustdoc JSON files created with mismatched feature config is not used
+    name: 'Caching with feature config changed'
+    runs-on: ubuntu-latest
+    needs:
+      - build-binary
+    steps:
+      - name: Checkout cargo-semver-checks
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          path: 'semver'
+
+      - name: Checkout fork of serde
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          repository: 'staniewzki/serde'
+          ref: 'c479da812fd421260667c02049e6d11edd82c2dc'
+          path: 'subject'
+
+      - name: Restore binary
+        id: cache-binary
+        uses: actions/cache/restore@v3
+        with:
+          path: bins/
+          key: bins-${{ runner.os }}-${{ github.run_id }}-${{ github.run_attempt }}
+          fail-on-cache-miss: true
+
+      - name: Install rust
+        id: toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Restore rustdoc
+        id: cache
+        uses: actions/cache/restore@v3
+        with:
+          key: cross-feature-caching-${{ runner.os }}-${{ steps.toolchain.outputs.cachekey }}-${{ hashFiles('semver/**/Cargo.lock') }}-${{ github.job }}-rustdoc
+          path: subject/target/semver-checks/cache
+
+      - name: Restore cargo index and rustdoc target dir
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: |
+            subject/target/semver-checks/local-serde-1_0_162
+
+      # Run even if there was a cache hit, to make sure the check passes.
+      - name: Run semver-checks
+        run: |
+          cd semver
+          ../bins/cargo-semver-checks semver-checks check-release --manifest-path="../subject/serde/Cargo.toml" --verbose
+
+      # Run cargo-semver-checks, enabling feature `unstable` this time.
```

```suggestion
      # Run cargo-semver-checks, enabling feature `unstable` this time.
      # We expect this to fail.
      # This run must not use the `serde` rustdoc without this feature as a baseline.
      # If that happens, this check will unexpectedly pass -- and the test will fail.
```

A bit more explanation like this would be good. Right now it's obvious to both of us what this does and why, but in 6 months it won't be so obvious anymore.

Also, I assume you've already checked that checking the `unstable` feature in current vs not checking `unstable` in baseline passes? Since that's what would happen if we accidentally incorrectly reused the cache, and we want to know that won't _also_ fail for some unrelated reason and thereby prevent this test from catching the problem.

---

Yes, I checked that!